st7735/st7735.go

```diff
@@ -0,0 +1,442 @@
+		d.rowOffset = 1
+	}
+	if cfg.ColumnOffset != 0 {
+		d.columnOffset = cfg.ColumnOffset
+	} else {
+		if d.model == MINI80x160 {
+			d.columnOffset = 26
+		} else {
+			d.columnOffset = 2
+		}
+	}
+
+	d.batchLength = d.width
+	if d.height > d.width {
+		d.batchLength = d.height
+	}
+	d.batchLength += d.batchLength & 1
+
+	// reset the device
+	d.resetPin.High()
+	time.Sleep(5 * time.Millisecond)
+	d.resetPin.Low()
+	time.Sleep(20 * time.Millisecond)
+	d.resetPin.High()
+	time.Sleep(150 * time.Millisecond)
+
+	// Common initialization
+	d.Command(SWRESET)
+	time.Sleep(150 * time.Millisecond)
+	d.Command(SLPOUT)
+	time.Sleep(500 * time.Millisecond)
+	d.Command(FRMCTR1)
+	d.Data(0x01)
+	d.Data(0x2C)
+	d.Data(0x2D)
+	d.Command(FRMCTR2)
+	d.Data(0x01)
+	d.Data(0x2C)
+	d.Data(0x2D)
+	d.Command(FRMCTR3)
+	d.Data(0x01)
+	d.Data(0x2C)
+	d.Data(0x2D)
+	d.Data(0x01)
+	d.Data(0x2C)
+	d.Data(0x2D)
+	d.Command(INVCTR)
+	d.Data(0x07)
+	d.Command(PWCTR1)
+	d.Data(0xA2)
+	d.Data(0x02)
+	d.Data(0x84)
+	d.Command(PWCTR2)
+	d.Data(0xC5)
+	d.Command(PWCTR3)
+	d.Data(0x0A)
+	d.Data(0x00)
+	d.Command(PWCTR4)
+	d.Data(0x8A)
+	d.Data(0x2A)
+	d.Command(PWCTR5)
+	d.Data(0x8A)
+	d.Data(0xEE)
+	d.Command(VMCTR1)
+	d.Data(0x0E)
+	d.Command(COLMOD)
+	d.Data(0x05)
+
+	if d.model == GREENTAB {
+		d.InvertColors(false)
+		d.Command(CASET)
+		d.Data(0x00)
+		d.Data(0x02)
+		d.Data(0x00)
+		d.Data(0x7F + 0x02)
+		d.Command(RASET)
+		d.Data(0x00)
+		d.Data(0x01)
+		d.Data(0x00)
+		d.Data(0x9F + 0x01)
+	} else if d.model == MINI80x160 {
+		d.isBGR = true
+		d.InvertColors(true)
+		d.Command(CASET)
+		d.Data(0x00)
+		d.Data(0x00)
+		d.Data(0x00)
+		d.Data(0x7F)
+		d.Command(RASET)
+		d.Data(0x00)
+		d.Data(0x00)
+		d.Data(0x00)
+		d.Data(0x9F)
+	}
+
+	// common color adjustment
+	d.Command(GMCTRP1)
+	d.Data(0x02)
+	d.Data(0x1C)
+	d.Data(0x07)
+	d.Data(0x12)
+	d.Data(0x37)
+	d.Data(0x32)
+	d.Data(0x29)
+	d.Data(0x2D)
+	d.Data(0x29)
+	d.Data(0x25)
+	d.Data(0x2B)
+	d.Data(0x39)
+	d.Data(0x00)
+	d.Data(0x01)
+	d.Data(0x03)
+	d.Data(0x10)
+	d.Command(GMCTRN1)
+	d.Data(0x03)
+	d.Data(0x1D)
+	d.Data(0x07)
+	d.Data(0x06)
+	d.Data(0x2E)
+	d.Data(0x2C)
+	d.Data(0x29)
+	d.Data(0x2D)
+	d.Data(0x2E)
+	d.Data(0x2E)
+	d.Data(0x37)
+	d.Data(0x3F)
+	d.Data(0x00)
+	d.Data(0x00)
+	d.Data(0x02)
+	d.Data(0x10)
+
+	d.Command(NORON)
+	time.Sleep(10 * time.Millisecond)
+	d.Command(DISPON)
+	time.Sleep(500 * time.Millisecond)
+
+	if cfg.Model == MINI80x160 {
+		d.Command(MADCTL)
+		d.Data(0xC0)
+	}
+
+	d.SetRotation(d.rotation)
+
+	d.blPin.High()
+}
+
+// Display does nothing, there's no buffer as it might be too big for some boards
+func (d *Device) Display() error {
+	return nil
+}
+
+// SetPixel sets a pixel in the screen
+func (d *Device) SetPixel(x int16, y int16, c color.RGBA) {
+	if x < 0 || y < 0 ||
+		(((d.rotation == NO_ROTATION || d.rotation == ROTATION_180) && (x >= d.width || y >= d.height)) ||
+			((d.rotation == ROTATION_90 || d.rotation == ROTATION_270) && (x >= d.height || y >= d.width))) {
```

Comment:
Wouldn't it be simpler to just compare against the result of `.Size()`? It seems to do the same thing.